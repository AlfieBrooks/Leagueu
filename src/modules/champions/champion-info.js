import React from 'react';
import {
  StyleSheet, Text, View, Image
} from 'react-native';
import FadeIn from 'react-native-fade-in-image';

import colourUtils from '../../utils/styles/colours';

export const ChampionInfo = ({
  championLevel,
  championPoints,
  championName,
  championImg,
}) => (
  <View style={styles.container}>
    <FadeIn placeholderStyle={styles.fadeImage}>
      <Image
        style={styles.image}
        source={{ uri: championImg }}
        resizeMode="contain"
      />
    </FadeIn>
    <Text style={styles.title}>{`${championName}`}</Text>
    <Text style={styles.text}>{`Level ${championLevel}`}</Text>
    <Text style={styles.text}>{`${championPoints} pts`}</Text>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    alignItems: 'center',
    paddingTop: 15,
    paddingBottom: 40,
    backgroundColor: colourUtils.seaBlue,
  },
  fadeImage: {
    backgroundColor: colourUtils.seaBlue,
  },
  image: {
    borderRadius: 35,
    height: 70,
    width: 70,
    marginBottom: 10,
  },
  title: {
    fontWeight: 'bold',
    marginBottom: 5,
    color: colourUtils.white,
  },
  text: {
    color: colourUtils.linkWater,
  },
});

export default ChampionInfo;
