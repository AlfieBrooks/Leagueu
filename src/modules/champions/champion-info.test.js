import React from 'react';
import renderer from 'react-test-renderer';

import { ChampionInfo } from './champion-info';

test('renders correctly', () => {
  const props = {
    championLevel: 5,
    championPoints: 500,
    championName: 'Test',
    championImg: 'img',
  };
  const wrapper = renderer.create(<ChampionInfo {...props} />).toJSON();
  expect(wrapper).toMatchSnapshot();
});
