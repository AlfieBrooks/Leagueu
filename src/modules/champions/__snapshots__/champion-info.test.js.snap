// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "paddingBottom": 40,
      "paddingTop": 15,
    }
  }
>
  <View
    placeholderStyle={
      Object {
        "backgroundColor": "#566270",
      }
    }
    useNativeDriver={true}
  >
    <Image
      onLoadEnd={[Function]}
      resizeMode="contain"
      source={
        Object {
          "uri": "img",
        }
      }
      style={
        Object {
          "borderRadius": 35,
          "height": 70,
          "marginBottom": 10,
          "width": 70,
        }
      }
    />
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 35,
              "height": 70,
              "marginBottom": 10,
              "width": 70,
            },
            Object {
              "backgroundColor": "#eee",
            },
            Object {
              "backgroundColor": "#566270",
            },
          ]
        }
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontWeight": "bold",
        "marginBottom": 5,
      }
    }
  >
    Test
  </Text>
  <Text
    style={
      Object {
        "color": "#F8FAFF",
      }
    }
  >
    Level 5
  </Text>
  <Text
    style={
      Object {
        "color": "#F8FAFF",
      }
    }
  >
    500 pts
  </Text>
</View>
`;
